ouroboros-network/src/Ouroboros/Network/Protocol/Trans/Hello/Type.hs

```diff
@@ -0,0 +1,130 @@
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE EmptyCase             #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE GADTs                 #-}
+{-# LANGUAGE InstanceSigs          #-}
+{-# LANGUAGE KindSignatures        #-}
+{-# LANGUAGE PolyKinds             #-}
+{-# LANGUAGE QuantifiedConstraints #-}
+{-# LANGUAGE ScopedTypeVariables   #-}
+{-# LANGUAGE TypeFamilies          #-}
+{-# LANGUAGE UndecidableInstances  #-}
+
+module Ouroboros.Network.Protocol.Trans.Hello.Type where
+
+import           Data.Void (Void)
+
+import           Network.TypedProtocol.Core
+import           Ouroboros.Network.Util.ShowProxy
+
+
+-- | 'Hello' protocol transformer.  It allows to transform a protocol
+-- which initial state is 'stIdle' for which the agancy might be held by
+-- the server, to a protocol with initial state 'StIdle' in which the client has
+-- agancy.  It adds 'MsgHello' which is a transition from 'StIdle' to
+-- @'StWrapped' stIdle@.
+--
+data Hello ps (stIdle :: ps) where
+    -- | 'StIdle' state is the initial state of the 'Hello' protocol.
+    --
+    StIdle  :: Hello ps stIdle
```

Comment:
Too many of these are called idle. Can we call this one the `StHello` state instead?